oak_io/src/handle.rs

```diff
@@ -82,17 +82,16 @@ impl From<Handle> for WriteHandle {
 /// }
 ///
 /// impl HandleVisit for Thing {
-///     fn visit<F: FnMut(&mut Handle)>(&mut self, mut visitor: F) -> F {
-///         visitor(&mut self.handle);
-///         visitor
+///     fn fold<B>(&mut self, init: B, mut f: fn(B, &mut Handle) -> B) -> B {
+///         f(init, &mut self.handle)
 ///     }
 /// }
 /// ```
 pub trait HandleVisit {
```

Comment:
Random thought, but would it make sense to just implement `Iterator` instead? Then we would get `fold` and other things for free. Conceptually it kind of makes sense for a type to return an iterator of handles, then the caller can do whatever it wants with it.

I guess this may be either `trait HandleVisit : Iterator<Item=Handle> {}`, or something with an extra layer of indirection, e.g. `trait HandleVisit { fn handles(&self) -> impl Iterator<Item=Handle> }` if we want to avoid clashes with `Iterator` that may already be on the type.

Reply:
It's an interesting idea! I wonder how that will work with recursion though, we might need some boxing there to make it work.

Note we would need `Iterator<Item=&mut Handle>` here, since we want to be able to modify the handles.